kolibri/plugins/coach/assets/src/utils/truncateSidePanelTitle.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Returns max number of characters allowed based on the provided window width.
+ *
+ * @param {number} windowWidth - The current window width
+ * @returns {number} The maximum allowed characters
+ */
+export function getMaxLengthByWidth(windowWidth) {
+  if (windowWidth <= 350) return 21;
```

Comment:
It works in most cases! But this will depend on the average width per character of a given language. For example, I just tested "Very long title in chinese" in chinese, and the close button is overflowed with this:

![image](https://github.com/user-attachments/assets/6603d791-5937-41e1-b810-4831e8c0efb8)

This can happen in other languages too. 

I just wonder if we can skip the case `Questions settings for 'Titl...'` with the quote at the end, and just have `Questions settings for 'Titl...` just truncating the last quote?

Reply:
Yes, you're right, it depends on the average width of each language. Truncating the title without a closing quotation mark felt incomplete and didn’t match the spec but the only way to do that is with overly complicated Javascript, so I went ahead and made the change to use only CSS instead.